mlflow/projects/__init__.py

```diff
@@ -91,19 +91,21 @@ def _run(uri, experiment_id, entry_point="main", version=None, parameters=None,
     Helper that delegates to the project-running method corresponding to the passed-in backend.
     Returns a ``SubmittedRun`` corresponding to the project run.
     """
+    if backend_name:
+        backend = loader.load_backend(backend_name)
+        if backend:
+            submitted_run = backend.run(run_id, experiment_id, uri, entry_point, parameters,
+                                        version, backend_config)
+            tracking.MlflowClient().set_tag(submitted_run.run_id, MLFLOW_PROJECT_BACKEND,
```

Comment:
Nit: to be safe, set the tag prior to calling `backend.run`? Just in case `backend.run` throws for whatever reason

Reply:
Here run_id can be None before the run method. Or backend implementation can create a new run_id. Therefore, to tag the run, the run_id must be the one from the submitted_run object.

Reply:
@smurching @jdlesage I was under the impression from prior conversation (https://github.com/mlflow/mlflow/pull/2566/files#r394638164) that we agreed to require the client to create the MLflow Run prior to beginning project backend execution and, therefore, that `run_id` must be defined prior to `backend.run()`. However, this dialogue seems to imply that the backend may be responsible for creating the run.

@smurching @jdlesage Please let me know if I am misinterpreting the outcome of https://github.com/mlflow/mlflow/pull/2566/files#r394638164. For consistency with preexisting behavior, I would prefer if the client be responsible for creating the MLflow run before beginning backend execution.

Reply:
Hi @dbczumar , Sorry, I misunderstood your first remark. 
The problem by creating the run in the client is the validation databrick test (this one: https://github.com/mlflow/mlflow/blob/master/tests/projects/test_databricks.py#L179) cannot be implemented.
In my first attempt, I created the runs in the client. But there are additionnal checks done by the databricks backend. If the backend_config is wrong, the run is not created to not "spam" the tracking server. To implement this behavior, it is needed that the databricks backend controls the run creation.
I don't have strong opinion about this. Could you confirm the client creates the run before to call the backend, please ? In this case, databricks projects will generate FAILED runs in the tracking server if the backend config is invalid.

Reply:
Ah, interesting...given this I'm wondering if we should reconsider forcing the client to create the run, the alternative would be to add a `validate_backend_config` method to the AbstractBackend API. Will sync with @dbczumar tomorrow on this & get back to you!

Reply:
Sync'd offline with @dbczumar on this, it seems reasonable to allow the backend to create runs. Could we update the backend interface to match that in flow 2 of [this gist](https://gist.github.com/smurching/6fe7368c6075ee3acf0220c672185b5e#file-project-backend-py-L57-L58)? Let me know if you have any questions, thanks!

Reply:
@smurching @jdlesage We should also include `tracking_uri` as part of this interface. (didn't see it in the gist).

Reply:
Sorry, I wrote the gist in a confusing way - the  top-level `def run` method definition in the gist is actually meant to mimic `mlflow.projects.run` (updated the gist to be clearer about that), hence why it doesn't accept a `tracking_uri` argument. To confirm:

* We should not add the `tracking_uri` argument to the user-facing `mlflow.projects.run` API (at least not in this PR)
* The `AbstractBackend.run` API called by `mlflow.projects.run` should accept a `tracking_uri` argument, as @dbczumar said.

Reply:
I did your last comment.
1/ AbstractBackend.run accepts a tracking_uri
2/ `tracking_uri` is not in the signature of `mlflow.projects.run`